src/AdvApi32.Desktop/AdvApi32.cs

```diff
@@ -888,14 +933,169 @@ public static partial class AdvApi32
         /// </returns>
         [DllImport(nameof(AdvApi32), SetLastError = true)]
         [return: MarshalAs(UnmanagedType.Bool)]
-        public static unsafe extern bool QueryServiceStatusEx(
+        public static extern bool QueryServiceStatusEx(
             SafeServiceHandle hService,
             SC_STATUS_TYPE InfoLevel,
-            void* lpBuffer,
+            IntPtr lpBuffer,
             int cbBufSize,
             out int pcbBytesNeeded);
 
         /// <summary>
+        /// Registers a function to handle service control requests.
+        /// </summary>
+        /// <param name="lpServiceName">
+        /// The name of the service run by the calling thread. This is the service name that the service control program specified in the CreateService function when creating the service.
+        /// If the service type is SERVICE_WIN32_OWN_PROCESS, the function does not verify that the specified name is valid, because there is only one registered service in the process.
+        /// </param>
+        /// <param name="lpHandlerProc">A reference to the handler function to be registered.</param>
+        /// <returns>If the function succeeds, the return value is a service status handle If the function fails, the return value is zero. </returns>
+        /// <remarks>
+        /// This function has been superseded by the <see cref="RegisterServiceCtrlHandlerEx"/> function.
+        /// A service can use either function, but the new function supports user-defined context data, and the new handler function supports additional extended control codes.
+        /// </remarks>
+        [DllImport(nameof(AdvApi32), CharSet = CharSet.Unicode)]
+        [Obsolete("Use LPHANDLER_FUNCTION_EX with RegisterServiceCtrlHandlerEx instead")]
```

Comment:
I just flagged as obsolete the functions that the MSDN documentation stated that was obsolete. Always describing the alternative on the message. I don't think we should mark with the obsolete attribute things that the documentation does not describe as obsolete.

I thought the obsolete warning is a nice way to tell the user that his code will work but that if he/she wants a proper support on future versions it should be better to update the code instead of suppressing the warning.

I don't know about Microsoft rules for deprecating Win 32 API functions but I have seen very little of it and only on functions that really don't make sense anymore (with newer/better alternatives on newer Windows versions and just present on unsupported Windows versions).